### PR TITLE
Create kubernetes_verify scenario, refactor test-go, verify jobs

### PR DESCRIPTION
**Pull request by fejta**
ref #1225 #1228 #1229 

Replace gotest-dockerized.sh, commands in test-go-foo.sh and commands in verify-foo.sh with a kubernetes_verify.py scenario.

Soon I will move scenario_args into a yaml file and replace test-go-foo.sh with test-go-foo.yaml once bootstrap supports this mode.

### CODE REVIEW

**Comment by fejta** [comment]
@k8s-bot bazel test this

**Comment by spxtr** [comment]
@k8s-bot bazel test this

**Comment by spxtr** [comment]
@k8s-bot bazel test this

**Comment by fejta** [comment]
Tests pass, ready for review

File: scenarios/kubernetes_verify.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    print >>sys.stderr, 'Run:', cmd
+    subprocess.check_call(cmd)
+
+
+def main(branch, script, force):
+    """Test branch using script, optionally forcing verify checks."""
+    verify_branch = re.match(r'.*(master|\d+\.\d+)', branch)
+    if not verify_branch:
+        raise ValueError(branch)
+    tag = VERSION_TAG[BRANCH_VERSION.get(verify_branch.group(1))]
```
Comment (krzyzacy) [inline scenarios/kubernetes_verify.py]: assume default need to be set? 

i.e.`BRANCH_VERSION.get(verify_branch.group(1),verify_branch.group(1))`?

File: scenarios/kubernetes_verify.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    remote = 'bootstrap-upstream'
+    uri = 'https://github.com/kubernetes/kubernetes.git'
+    try:
+        check('git', 'remote', 'remove', remote)
+    except subprocess.CalledProcessError:
+        pass
+    check('git', 'remote', 'add', remote, uri)
+    check('git', 'remote', 'set-url', '--push', remote, 'no_push')
+    # If .git is cached between runs this data may be stale
+    check('git', 'fetch', remote)
```
Comment (krzyzacy) [inline scenarios/kubernetes_verify.py]: jobs like `pull-kubernetes-unit.sh` does not have those git commands with them, should those be conditional?

Comment (fejta) [inline scenarios/kubernetes_verify.py]: Seems easier/simpler to just add this everywhere to me?

File: scenarios/kubernetes_verify.py (hunk 1 above)
Comment (fejta) [inline scenarios/kubernetes_verify.py]: Done

File: .travis.yml [hunk 3]
```diff
@@ -30,6 +30,8 @@ install:
 script:
     - ./verify/verify-boilerplate.py
     - ./verify/verify-bazel.sh
+    - pylint scenarios/*.py
+    - pylint jenkins/bootstrap.py  # TODO(fejta): all python files
```
Comment (krzyzacy) [inline .travis.yml]: duplicated?

Comment (fejta) [inline .travis.yml]: Done

File: scenarios/kubernetes_verify.py (hunk 2 above)
Comment (krzyzacy) [inline scenarios/kubernetes_verify.py]: as long as it's not causing troubles

**Pull request merged**